### DISCUSSION TITLE
Timer trigger not working!

### DISCUSSION THREAD

**Discussion by Let-Me-Adam** [microsoft/hack-together-teams#76] [Q&A]
Hello,

We are trying to create an app that notify's the user at 30 minute intervals. We are having problems getting the timer trigger to activate and send an adaptive card to the chat and wondered if anyone would be able to give some advice on what we might be missing. At the moment we are using the template for the time trigger bot which hasn't been sending any notifications.  

We are both new to developing on VS Code ad the Teams Toolkit and wanted to use this as a learning opportunity. Any in depth explanation for beginners would be MUCH appreciated!

If you need any more info about the project, please let us know. 

Thanks.
Best wises,
Adam and Monika

**Comment by svrooij** [comment]
@Let-Me-Adam you're talking about the `timer trigger`, do you mean the [Timer Trigger for Azure Functions](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer)? I was faced with a similar issue some time ago, and it had something to do with the App service where the Timer was hosted not being set to `AlwaysOn`. More details can be found [here](https://github.com/Azure/azure-functions-host/wiki/Investigating-and-reporting-issues-with-timer-triggered-functions-not-firing)

**Reply by Let-Me-Adam** [reply]
Thank you! Will give this solution a go and let you know if it works 😊

**Comment by aycabas** [comment]
Hi @Let-Me-Adam, you can also try to setup `RunOnStartup` that helps you run the trigger when you start the app. It's not recommended to use in production but it's great for testing for sure. The only thing you need to do is to set up `RunOnStartup` on your local.settings.json. 

Here is the documentation: https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=python-v2%2Cin-process&pivots=programming-language-csharp#attributes

Here is sample local.settings.json: https://learn.microsoft.com/en-us/azure/azure-functions/functions-develop-local#local-settings-file

Please let me know if it works or if you need further support.

**Reply by Let-Me-Adam** [reply]
Amazing! Will look through the links and update you with what happens. Thanks so much! 😁
